src/PeerReview/Web/API.hs

```diff
@@ -1,52 +1,80 @@
     ) where
 
-import           Control.Monad.IO.Class (liftIO)
+import           Control.Arrow             ((&&&))
+import           Control.Monad             (unless)
+import           Control.Monad.IO.Class    (liftIO)
+import           Data.Map                  (Map)
+import qualified Data.Map                  as M
+import           Data.Text                 (Text)
+import           Network.HTTP.Types.Status (status404)
 import           Web.Spock.Shared
 
 import           PeerReview.Core
+import           PeerReview.Types          (PeerReview (..), PeerReviewID,
+                                            ReviewFilter (..))
 import           PeerReview.Web.Types
 
--- Give a task to review if possible.
-new :: Action ctx a
-new = do
-    email <- param' "email"
+create :: Action ctx a
+create = do
+    email <- crbUid <$> jsonBody'
     e <- asEnv <$> getState
     eReview <- liftIO (findTaskToReview e email)
-    either json json eReview
+    maybe json404 json (reviewResponse <$> eReview)
 
--- Create peer reviews.
-review :: Action ctx a
-review = do
-    r <- jsonBody'
-    json =<< liftIO (createReview r)
+update :: Int -> Action ctx a
+update rid = do
+    env        <- asEnv <$> getState
+    reviewData <- (urbComment &&& urbScore) <$> jsonBody'
+    success    <- liftIO (updateReview env (fromIntegral rid) reviewData)
+    unless success (setStatus status404)
+    json success
 
--- List peer reviews for user.
+-- Find all reviews when there are no params.
+-- When params are present pick correct filter or return 404.
 list :: Action ctx a
 list = do
-    email <- param' "email"
-    e <- asEnv <$> getState
-    json =<< liftIO (listReviewsForUser e email)
+    ps <- params
+    e  <- asEnv <$> getState
+    if length ps < 1 then
+        json =<< liftIO (fmap reviewResponse <$> listReviews e Nothing) else
+        case pickFilter ps of
+            Nothing -> json404
+            p       -> json =<< liftIO (fmap reviewResponse <$> listReviews e p)
 
--- List completed reviews.
-completed :: Action ctx a
-completed = do
-    e <- asEnv <$> getState
-    json =<< liftIO (completedReviews e)
+find :: Int -> Action ctx a
+find rid = do
+    env     <- asEnv <$> getState
+    eResult <- liftIO (findReview env (fromIntegral rid))
+    maybe json404 json (reviewResponse <$> eResult)
+
+
+-- Return False as JSON with status 404.
+json404 :: Action ctx a
+json404 = setStatus status404 >> json False
+
+-- Turn Review data into API format.
+reviewResponse :: (PeerReviewID,PeerReview) -> ReviewResponse
+reviewResponse (rid, PeerReview sid tid sc c score revid status) =
+    ReviewResponse rid sid tid sc c score revid status
 
--- Mark review as accepted.
-accept :: Action ctx a
-accept = do
-    r <- jsonBody'
-    json =<< liftIO (acceptReview r)
+-- All available filters for review listing.
+filters :: Map Text (Text -> ReviewFilter Text)
+filters = M.fromList
+    [ ("filter[task]",     ByTask)
+    , ("filter[reviewer]", ByReviewer)
```

Comment:
We should have a `ByReviewee` filter as well. This might lead to a database change as well (storing the `reviewee` information in our database). Separate PR?